projects/ancestor/ancestor.py

```diff
@@ -0,0 +1,111 @@
+class Queue():
+    def __init__(self):
+        self.queue = []
+    def enqueue(self, value):
+        self.queue.append(value)
+    def dequeue(self):
+        if self.size() > 0:
+            return self.queue.pop(0)
+        else:
+            return None
+    def size(self):
+        return len(self.queue)
+
+class Graph:
+    """Represent a graph as a dictionary of vertices mapping labels to edges."""
+    def __init__(self):
+        self.vertices = {}
+    def add_vertex(self, vertex_id):
+        if vertex_id not in self.vertices:
+            self.vertices[vertex_id] = set()
+    def add_edge(self, v1, v2):
+        if v1 in self.vertices and v2 in self.vertices:
+            self.vertices[v1].add(v2)
+        else:
+            raise IndexError("That vertex does not exist!")
+
+
+
+def earliest_ancestor(ancestors, starting_node):
+    # Build the graph
```

Comment:
Nice working solution for the earliest ancestor problem.

I like how you added a small amount of comments where necessary.  This helps to keep the reader on track of what is going on as they try to digest the code, but does not become overwhelming as too many comments may do at times.